WalletWasabi.Gui/Controls/LockScreen/SlideLock.cs

```diff
@@ -0,0 +1,187 @@
+using Avalonia;
+using Avalonia.Animation;
+using Avalonia.Animation.Easings;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Styling;
+using Avalonia.Utilities;
+using System;
+
+namespace WalletWasabi.Gui.Controls.LockScreen
+{
+	public class SlideLock : ContentControl
+	{
+		private Thumb _thumb;
+		private Grid _container;
+		private bool _isLocked;
+		private Animation _closeAnimation;
+		private Animation _openAnimation;
+		private bool _canSlide;
+		private double _threshold;
+
+		static SlideLock()
+		{
+			AffectsArrange<SlideLock>(ValueProperty);
+		}
+
+		public SlideLock()
+		{
+			this.GetObservable(ValueProperty)
+				.Subscribe(x => Opacity = x / 100);
+
+			_closeAnimation = new Animation
+			{
+				Duration = TimeSpan.FromSeconds(1),
+				Easing = new QuarticEaseInOut(),
+				FillMode = FillMode.Both,
+				Children =
+				{
+					new KeyFrame
+					{
+						Setters =
+						{
+							new Setter
+							{
+								Property = ValueProperty,
+								Value = 100d
+							}
+						},
+						Cue = new Cue(1d)
+					}
+				}
+			};
+
+			_openAnimation = new Animation
+			{
+				Duration = TimeSpan.FromSeconds(1),
+				Easing = new QuarticEaseInOut(),
+				FillMode = FillMode.Both,
+				Children =
+				{
+					new KeyFrame
+					{
+						Setters =
+						{
+							new Setter
+							{
+								Property = ValueProperty,
+								Value = 0d
+							}
+						},
+						Cue = new Cue(1d)
+					}
+				}
+			};
+
+			this.GetObservable(IsLockedProperty)
+				.Subscribe(async x =>
+				{
+					if (x)
+					{
+						await _closeAnimation.RunAsync(this);
+					}
+					else
+					{
+						await _openAnimation.RunAsync(this);
+					}
+				});
```

Comment:
Give `x` a more descriptive name. What about something like this?


```suggestion
var animation = isLocked ? _closeAnimation : _openAnimation;
await animation.RunAsync(this);
```

Reply:
done